include/dr/mhp/halo.hpp

```diff
@@ -315,6 +315,16 @@ template <typename T, typename Memory = default_memory<T>> class span_group {
 };
 
 struct halo_bounds {
+  halo_bounds(std::size_t bound_ = 0, bool periodic_ = false) {
+    prev = next = bound_;
+    periodic = periodic_;
+  }
+  halo_bounds(std::size_t prev_, std::size_t next_, bool periodic_ = false) {
```

Comment:
please remove default values from ctors

int is convertible to bool - when one writes halo_bounds(1,0) it is not clear for the compiler if we use first or second ctor